homeassistant/components/climate/__init__.py

```diff
@@ -948,6 +990,54 @@ async def async_service_temperature_set(
         else:
             kwargs[value] = temp
 
+    if not hasattr(entity, "min_temperature_range"):
+        await entity.async_set_temperature(**kwargs)
+        return
+
+    if (
+        (min_temp_range := entity.min_temperature_range)
+        and (target_low_temp := kwargs.get(ATTR_TARGET_TEMP_LOW))
+        and (target_high_temp := kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        and (target_high_temp - target_low_temp) < min_temp_range
+    ):
+        # Ensure target_low_temp is not higher than target_high_temp.
+        if target_low_temp > target_high_temp:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="low_temp_higher_than_high_temp",
+            )
+
+        # Ensure deadband between target temperatures.
+        initial_low_temp = target_low_temp
+        if (target_high_temp - target_low_temp) < min_temp_range:
+            target_low_temp = target_high_temp - min_temp_range
+
+        # Guard target_low_temp isn't lower than entity low_temp
+        if target_low_temp < min_temp:
+            raise ServiceValidationError(
```

Comment:
I think the behavior should be symmetric; if we try to "fix" the service call by adjusting the passed in low_temp, we should try to adjust the high_temp too.

If targetting multiple termostats is a valid use case, it might make sense to allow the user to pass in a narrow deadband, and then adjust it in the service call for thermostats which support it?

Is there a frontend design, or even better a draft PR, which takes the `min_temp_range` into account?

Reply:
Made it symmetric now in the sense that it first moves the `low_temp` and second the `high_temp` if needed.

It does not check if the deadband makes the equation impossible e.g. in theory an endless loop to get inside min/max temp and still ensure the deadband. Is such check necessary and should raise?

Reply:
If it's targeting multiple entities it would seem strange if the user is allowed to set a deadband as the integrations should be in control of that?